Gifski/VideoValidator.swift

```diff
@@ -184,6 +184,47 @@ struct VideoValidator {
 			return .failure
 		}
 
-		return .success(newAsset, newVideoMetadata)
+		// Find first non-empty frame
+		do {
+			let reader = try AVAssetReader(asset: newAsset)
+			let readerOutput = AVAssetReaderTrackOutput(track: newAsset.firstVideoTrack!, outputSettings: nil)
+			reader.add(readerOutput)
+			reader.startReading()
+
+			while let sampleBuffer = readerOutput.copyNextSampleBuffer() {
```

Comment:
Per docs, you need to check the `.status` property, and handle the different cases properly.

Reply:
Fixed in 9455603.